### PR TITLE
Fix I2s async-tx

### LINKED ISSUE #1832: I2S write broken
Apparently i2s-write is broken - luckily released 0.19.0 is not affected

```
❯ git bisect good
6fd9443c138bc3eb46da5dead333ccf6d424cec7 is the first bad commit
commit 6fd9443c138bc3eb46da5dead333ccf6d424cec7
Author: Dominic Fischer <14130965+Dominaezzz@users.noreply.github.com>
Date:   Tue Jul 16 09:52:48 2024 +0100

    Allow users to easily name DMA channels (#1771)

    * Introduce public DmaChannel trait

    * Use DmaChannel trait in all peripherals

    * Simplify ChannelTypes trait

    * changelog

    * Rename things

    * missed a spot

    ---------

    Co-authored-by: Dominic Fischer <git@dominicfischer.me>

 esp-hal/CHANGELOG.md             |   1 +
 esp-hal/src/aes/mod.rs           |  24 +++---
 esp-hal/src/dma/gdma.rs          |  33 +++++---
 esp-hal/src/dma/mod.rs           | 168 ++++++++++++++++++---------------------
 esp-hal/src/dma/pdma.rs          |  20 +++--
 esp-hal/src/i2s.rs               |  88 ++++++++++----------
 esp-hal/src/lcd_cam/cam.rs       |  25 +++---
 esp-hal/src/lcd_cam/lcd/i8080.rs |  29 +++----
 esp-hal/src/parl_io.rs           |  84 ++++++++++----------
 esp-hal/src/spi/master.rs        |  68 ++++++++--------
 esp-hal/src/spi/slave.rs         |  36 +++++----
 11 files changed, 291 insertions(+), 285 deletions(-)

```

Easy to reproduce via `cargo xtask run-example examples esp32c6 embassy_i2s_sound` - after that commit it doesn't do anything anymore

This is especially bad since the hil-test didn't catch this 😢

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Fixes #1832 

#### Testing
Check with `cargo xtask run-example examples esp32c6 embassy_i2s_sound`

### CODE REVIEW

**Comment by MabezDev** [comment]
Could we add an async i2s test while we're here?

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

File: hil-test/tests/i2s_async.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    #[test]
+    fn run_test() {
+        unsafe fn __make_static<T>(t: &mut T) -> &'static mut T {
+            ::core::mem::transmute(t)
+        }
+
+        let mut executor = esp_hal_embassy::Executor::new();
+        let executor = unsafe { __make_static(&mut executor) };
+        executor.run(|spawner| {
```
Comment (MabezDev) [inline hil-test/tests/i2s_async.rs:167]: If we're doing all this to get the spawner, I think we can just use https://docs.embassy.dev/embassy-executor/git/cortex-m/struct.Spawner.html#method.for_current_executor to get the spawner?

**Review by MabezDev — CHANGES_REQUESTED** [review]

File: hil-test/tests/i2s_async.rs (hunk 1 above)
Comment (bjoernQ) [reply hil-test/tests/i2s_async.rs:167]: Ah I was looking for something like that but didn't find it 👍 will try

**Review by MabezDev — APPROVED** [review]
Nice! Thanks for adding the test case :heart:

**Comment by MabezDev** [comment]
Could you rebase on main to pull in the latest CI changes?

**Comment by jessebraham** [comment]
> Could you rebase on main to pull in the latest CI changes?

The merge queue should squash and rebase this on `main` already, shouldn't it?

**Pull request merged**